packages/@tinacms/schema-tools/src/schema/addNamespaceToSchema.test.ts

```diff
@@ -0,0 +1,352 @@
+        },
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    // Verify that namespaces are correctly constructed
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'],
+          nested: {
+            name: 'child',
+            namespace: ['child', 'child'], // Should reflect the nesting with repeated name
+          },
+        },
+      ],
+    })
+
+    // Ensure nested namespaces are correctly formed
+    expect((result.children[0] as Node).namespace).toEqual(['child'])
+    expect((result.children[0].nested as Node).namespace).toEqual([
+      'child',
+      'child',
+    ])
+  })
+
+  test('should preserve existing namespaces for repeated names at different levels', () => {
+    const node = {
+      name: 'root',
+      namespace: [], // Existing namespace
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'], // Existing namespace
+          nested: {
+            name: 'child',
+            namespace: ['child', 'child'], // Existing namespace for repeated name
+          },
+        },
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    // Verify that the function does not modify the existing namespaces
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'],
+          nested: {
+            name: 'child',
+            namespace: ['child', 'child'],
+          },
+        },
+      ],
+    })
+
+    // Ensure the namespaces are correctly preserved
+    expect((result.children[0] as Node).namespace).toEqual(['child'])
+    expect((result.children[0].nested as Node).namespace).toEqual([
+      'child',
+      'child',
+    ])
+  })
+
+  test('should preserve existing namespaces and not modify them', () => {
+    const node = {
+      name: 'root',
+      namespace: [], // Assume namespace is already set
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'],
+          nested: {
+            name: 'grandchild',
+            namespace: ['child', 'grandchild'], // Correct namespace already set
+          },
+        },
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    // Verify that the function does not modify the existing namespaces
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'],
+          nested: {
+            name: 'grandchild',
+            namespace: ['child', 'grandchild'],
+          },
+        },
+      ],
+    })
+
+    // Ensure the namespaces are not modified
+    expect(result).toStrictEqual(node) // Check if the function returns the same object
+  })
+
+  test('deeply nested schema issue #5364', () => {
```

Comment:
Test "should preserve existing namespaces and not modify them" says that namespaces should not be modified once created. Does that then mean this test is doing nothing since they can not be modified?

Reply:
The bug involves passing a schema with namespace already on it and the namespace getting modified. There is a node with this namespace:

```
[
    "page",
    "blocks",
    "generativeImageBlock",
    "advancedImageOptions",
    "effects",
    "optimizations",
    "improve"
]
```

That gets modified to:

```
[
    "page",
    "blocks",
    "imageBlock",
    "advancedImageOptions",
    "effects",
    "optimizations",
    "improve"
]
```

So the test is validating that the namespace doesn't get modified